### ISSUE TITLE
Java 11 module compatibility

### ISSUE THREAD

**Issue by abyrd** [conveyal/kryo-tools#3]
I added a module-info.java in 8152211cc5a883aba0f92087bad4117d4c86eb0c,  however during the build Maven warns us not to publish any artifact based on automatic filename based module names. This project still has three:

```
[WARNING] *******************************************************************************************************************************************************************************
[WARNING] * Required filename-based automodules detected: [kryo-4.0.2.jar, trove4j-3.0.3.jar, slf4j-api-1.7.25.jar]. Please don't publish this project to a public artifact repository! *
[WARNING] *******************************************************************************************************************************************************************************
```

The new Kryo 5 release will declare a module name. New SLF4J versions already have module names. The tricky one is Trove, which no longer has regular releases. Someone did a 3.1 release
recently under the new name net.sf.trove4j:core but I can only get this to build by requiring the module "core" which is strange - are they claiming this module name at the highest level of the namespace?

We need to start at the leaf nodes of the dependency tree and work up to release all modular artifacts. Unfortunately it seems like projects are migrating very
slowly to named modules: https://blog.frankel.ch/hard-look-state-java-modularization/

**Comment by abyrd** [comment]
I now better understand the above mentioned "Please don't publish this project to a public artifact repository! " after reading discussion of Jigsaw modules from the heads of Sonatype and Maven: https://dzone.com/articles/java-automodules-considered-bad-for-your-health Here and in other articles they plead for people to completely avoid fully-automatic filename based automodules. 

This post: https://blog.joda.org/2017/05/java-se-9-jpms-automatic-modules.html comes to the conclusion that you should not upload an artifact with a module-info.java to a Maven repository unless all the required modules in that module-info.java also declare a module name, via module-info.java or at least via `Automatic-Module-Name` in the JAR manifest.

So we should only include the `Automatic-Module-Name` in the JAR manifest until our dependencies all specify a module name, then we can add a module-info.java. The question is then how we would build the library at all under Java 11 without a module-info.java.

Naming: They suggest all module names should be namespaced and probably based on Maven group+artifact coordinates. Generally, modules should be named in reverse-DNS style like Java packages, and ideally the name of the module is the shortest common prefix of all the packages it exports.

**Comment by abyrd** [comment]
Addressed in v1.5.0.
